test/arrayops.jl

```diff
@@ -3,8 +3,13 @@
 # Array test
 isdefined(Main, :OffsetArrays) || @eval Main include("testhelpers/OffsetArrays.jl")
 using .Main.OffsetArrays
+
 using SparseArrays
 
+if !isdefined(@__MODULE__, :T24Linear)
+     include("testhelpers/arrayindexingtypes.jl")
+ end
```

Comment:
I copied this from #35298, not sure if it's needed at all, nor whether it should be written like `isdefined || @eval` just above. (And at very least a space needs fixing, sorry.)

Reply:
Changed in https://github.com/JuliaLang/julia/pull/35304/commits/a524b230febe94c99ab3dcbbd3917fc13a171576, I was hoping CI would tell me if that's OK, but it's failed on some unrelated unicode thing instead.

Reply:
I think this is needed, and have restored it to something like what I had just above. 

CI is failing, I don't completely see why, but I don't think caused by this.